inspirehep/modules/multieditor/actions.py

```diff
@@ -0,0 +1,282 @@
+
+    def apply_action(self, record, schema, position=0, checked=0):
+        """Recursive function to update a record primitive key."""
+        new_schema, checked, key, condition_failed = self.progress_keys(record, schema, position, checked)
+        if condition_failed:
+            return
+        if not record.get(key):
+            return
+        if position == len(self.keys) - 1:
+            if isinstance(record[key], list):
+                if self.match_type == 'regex':
+                    record[key] = [self.value if re.search(
+                        re.escape(self.value_to_check),
+                        x) else x for x in record[key]]
+
+                elif self.match_type == 'equal':
+                    record[key] = [self.value if x == self.value_to_check else x for x in record[key]]
+
+                elif self.match_type == 'contains':
+                    record[key] = [self.value if self.value_to_check in x else x for x in record[key]]
+                self.changed = True
+            else:
+                if self.match_type == 'equal' and record[key] == self.value_to_check:
+                    record[key] = self.value
+                if self.match_type == 'regex' and re.search(
+                        re.escape(self.value_to_check),
+                        record[key]):
+                            record[key] = self.value
+                elif self.match_type == 'contains' and self.value_to_check in record[key]:
+                    record[key] = self.value
+                self.changed = True
+                return
+        else:
+            if isinstance(record[key], list):
+                for array_record in record[key]:
+                    self.apply_action(array_record, new_schema, position + 1, checked)
+            else:
+                self.apply_action(record[key], new_schema, position + 1, checked)
+
+
+def create_schema_record(schema, path, value):
+    """Object creation in par with the schema."""
+    record = {}
+    temp_record = record
+    new_schema = schema
+    if new_schema['type'] == 'array':
+        new_schema = new_schema['items']['properties']
+    elif new_schema['type'] == 'object':
+        new_schema = new_schema['properties']
+    for key in path:
+        new_schema = new_schema[key]
+        if new_schema['type'] == 'object':
+            new_schema = new_schema['properties']
+            temp_record[key] = {}
+            temp_record = temp_record[key]
+
+        elif new_schema['type'] == 'array':
+            if new_schema['items']['type'] == 'object':
+                new_schema = new_schema['items']['properties']
+            if len(path) == 1:
+                temp_record[key] = [value]
+                return record
+            else:
+                temp_record[key] = [{}]
+                temp_record = temp_record[key][0]
+    temp_record[path[-1]] = value
+    return record
+
+
+def check_value(record, match_type, keys, value_to_check, position):
```

Comment:
Are there tests for this function to make sure it is doing the correct thing?

Reply:
Yes this function is being run for every condition, so all the conditions in the tests are triggering it.